production/helm/loki/templates/querier/hpa.yaml

```diff
@@ -1,4 +1,7 @@
 {{- $isDistributed := eq (include "loki.deployment.isDistributed" .) "true" -}}
+{{- if and .Values.querier.kedaAutoscaling.enabled .Values.querier.autoscaling.enabled}}
+{{- fail "Cannot enable both HPA and Keda based autoscaling at the same time."}}
```

Comment:
What do you think about not duplicating this fail in both HPA and the ScaledObject? 
Could we somehow either add it here https://github.com/grafana/loki/blob/31fcc816e41cab894b7aeae8093d8efcac909d78/production/helm/loki/templates/validate.yaml or in a validate.yaml file in the querier folder?